traitsui/tests/editors/test_check_list_editor.py

```diff
@@ -315,92 +315,78 @@ def test_custom_editor_mapping_name_tuple(self):
             self.check_checklist_mappings_tuple_name_change("custom")
 
     def test_simple_editor_checklist_values_change_dispose(self):
-        if is_current_backend_wx():
-            # Missing cleanup in wx (enthought/traitsui#752)
-            with self.assertRaises(AttributeError):
-                self.check_checklist_values_change_after_ui_dispose("simple")
-        else:
-            self.check_checklist_values_change_after_ui_dispose("simple")
+        self.check_checklist_values_change_after_ui_dispose("simple")
 
     def test_custom_editor_checklist_values_change_dispose(self):
-        if is_current_backend_wx():
-            # Missing cleanup in wx (enthought/traitsui#752)
-            with self.assertRaises(AttributeError):
-                self.check_checklist_values_change_after_ui_dispose("custom")
-        else:
-            self.check_checklist_values_change_after_ui_dispose("custom")
+        self.check_checklist_values_change_after_ui_dispose("custom")
 
 
 @skip_if_null
 class TestSimpleCheckListEditor(unittest.TestCase):
 
+    @contextlib.contextmanager
     def setup_gui(self, model, view):
-        gui = GUI()
-        ui = model.edit_traits(view=view)
-        self.addCleanup(ui.dispose)
-
-        gui.process_events()
-        editor = ui.get_editors("value")[0]
-        combobox = editor.control
-
-        return gui, editor, combobox
+        with create_ui(model, dict(view=view)) as ui:
```

Comment:
Agreed. I was thinking this would be an orthogonal change that causes no change to the test logic, I guess it would be fine to include here as it would not be too hard to see that it causes no logical change.

Reply:
As soon as I tried, I realized that *is* going to introduce more differences into the test logic because some of these instance methods have `GUI.process_events` and some don't. I would create separate functions for them but then that's kind of defeating the point of reducing repetition. Abort... sorry!

Reply:
Up to you then

Reply:
Thanks. I think I will try in a separate PR to remove these process_events after the UI is created, then we will know whether they are generally needed, and then I can refactor these functions.